h2o-py/h2o/frame.py

```diff
@@ -2633,8 +2633,14 @@ def rbind(self, data):
             if frame.ncol != self.ncol:
                 raise H2OValueError("Cannot row-bind a dataframe with %d columns to a data frame with %d columns: "
                                     "the columns must match" % (frame.ncol, self.ncol))
-            if frame.columns != self.columns or frame.types != self.types:
-                raise H2OValueError("Column names and types must match for rbind() to work")
+            if frame.columns != self.columns:
+                raise H2OValueError("Column names must match for rbind() to work")
+            if frame.types != self.types:
+                validTypes = [u'float', u'real', u'double', u'int', u'long', u'numeric']
+                for eachKey in frame.types.keys():
+                    if not(frame.types[eachKey] in validTypes) or not(self.types[eachKey] in validTypes):
```

Comment:
Michalk:

My fix only changes for numerical types.  It did not change the original implementation.  I think you are correct though.  Do you want to be able to bind strings and categoricals?  I can easily add that.  Thanks, Wendy

Reply:
if types are not the same (frame.types != self.types) due to a numerical inconsistency, you compare types for all columns (eachKey in frame.types.keys()) and require it to be in one of the "valid types".

in the previous implementation if there would be no issue with categorical values (if the numerical types were the same)

Can you add a categorical column to your test?

Reply:
@michalkurka:  Still confused about what your concern is though.  I can easily add a categorical column to the frame.  However this will not answer your question about whether a categorical column can rbind with a string column.  Do you want me to test that?  Or just add a categorical column to the frame and see if they can rbind?  Let me know.  Wendy

Reply:
@michalkurka 

I added the categorical column to the test.  I am keeping the first part of the test unchanged since with two columns, calling summary will not change the column type from real to int.

Reply:
@wendycwong it seems like we don't understand each other here, let me demonstrate the problem with a test:

```
def test_rbind_summary():
    df1 = h2o.H2OFrame([[1,"a"],[2,"b"]])
    df1[1]=df1[1].asfactor()

    df2 = h2o.H2OFrame([[2.2,"b"],[1.1,"a"]])
    df2[1]=df2[1].asfactor()
    
    print(df1.summary())
    print(df2.summary())
    
    df3 = df1.rbind(df2)
    
    print(df3)
```

This code fails on 
```
h2o.exceptions.H2OValueError: Column types must match for rbind() to work.  First column type enum.  Second column type enum)
```

This happens when `df1.types != d2.types`. In that case, your code runs a second check but this check only considers numerical values and expects all the types to be numeric.

Reply:
Fixed.